Simplified/AxisService/NYPLAxisTasksSynchnorizer.swift

```diff
@@ -0,0 +1,73 @@
+//
+//  NYPLAxisTasksAggregator.swift
+//  Open eBooks
+//
+//  Created by Raman Singh on 2021-06-08.
+//  Copyright © 2021 NYPL Labs. All rights reserved.
+//
+
+import Foundation
+
+protocol NYPLAxisTasksSynchorizing: class {
+  func startSynchronizedTask()
+  func endSynchronizedTask()
+  func waitForSynchronizedTaskToFinish()
+  func startSynchronizedTaskWhenIdle()
+  func runOnCompletingAllSynchronizedTasks(on queue: DispatchQueue, closure: (() -> Void)?)
+}
+
+extension NYPLAxisTasksSynchorizing {
+  func startSynchronizedTaskWhenIdle() {
+    waitForSynchronizedTaskToFinish()
+    startSynchronizedTask()
+  }
+}
+
+/// Wrapper class for aggregating a set of tasks and synchronizing behavior
+class NYPLAxisTasksSynchnorizer: NYPLAxisTasksSynchorizing {
```

Comment:
so, on a deeper look, this class is actually kind of synonym of DispatchGroup. It wraps it with equivalent apis. Did you add it mainly to add more appropriate nouns and verbs that match the objective better?